test/jdk/java/io/File/GetAbsolutePath.java

```diff
@@ -22,74 +22,89 @@
+import java.util.stream.Stream;
 
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.*;
 
 public class GetAbsolutePath {
 
-    private static boolean ignoreCase = false;
+    private static final String USER_DIR = System.getProperty("user.dir");
 
-    private static void ck(String path, String ans) throws Exception {
-        File f = new File(path);
-        String p = f.getAbsolutePath();
-        if ((ignoreCase && p.equalsIgnoreCase(ans)) || p.equals(ans))
-            System.err.println(path + " ==> " + p);
-        else
-            throw new Exception(path + ": expected " + ans + ", got " + p);
+    private static char driveLetter() {
+        assert System.getProperty("os.name").startsWith("Windows");
+
+        if ((USER_DIR.length() > 2) && (USER_DIR.charAt(1) == ':')
+            && (USER_DIR.charAt(2) == '\\'))
+            return USER_DIR.charAt(0);
+
+        throw new RuntimeException("Current directory has no drive");
     }
 
-    private static void testWin32() throws Exception {
-        String wd = System.getProperty("user.dir");
-        char d;
-        if ((wd.length() > 2) && (wd.charAt(1) == ':')
-            && (wd.charAt(2) == '\\'))
-            d = wd.charAt(0);
-        else
-            throw new Exception("Current directory has no drive");
-        ck("/foo/bar", d + ":\\foo\\bar");
-        ck("\\foo\\bar", d + ":\\foo\\bar");
-        ck("c:\\foo\\bar", "c:\\foo\\bar");
-        ck("c:/foo/bar", "c:\\foo\\bar");
-        ck("\\\\foo\\bar", "\\\\foo\\bar");
+    private static Stream<Arguments> windowsSource() {
+        char drive = driveLetter();
+        return Stream.of(Arguments.of("/foo/bar", drive + ":\\foo\\bar"),
+                         Arguments.of("\\foo\\bar", drive + ":\\foo\\bar"),
+                         Arguments.of("c:\\foo\\bar", "c:\\foo\\bar"),
+                         Arguments.of("c:/foo/bar", "c:\\foo\\bar"),
+                         Arguments.of("\\\\foo\\bar", "\\\\foo\\bar"),
+                         Arguments.of("", USER_DIR), // empty path
+                         Arguments.of("\\\\?\\foo", USER_DIR + "\\foo"),
+                         Arguments.of("\\\\?\\C:\\Users\\x", "C:\\Users\\x"),
+                         Arguments.of("\\\\?\\" + drive + ":", USER_DIR),
+                         Arguments.of("\\\\?\\" + drive + ":bar", USER_DIR + "\\bar"));
+    }
+
+    @EnabledOnOs(OS.WINDOWS)
+    @ParameterizedTest
+    @MethodSource("windowsSource")
+    public void windows(String path, String absolute) throws IOException {
+        assertEquals(absolute.toLowerCase(),
+                     new File(path).getAbsolutePath().toLowerCase());
```

Comment:
WinNTFileSystem.compareTo uses compareToIgnoreCase so this test should probably do the same.

Reply:
So changed in 2f31318530d7093dc69c07115c5d46b63612ff4b.